app/controllers/products_controller.rb

```diff
@@ -0,0 +1,45 @@
+class ProductsController < ApplicationController
+
+  def new
+    @vendor_id = params[:vendor_id]
+    @product = Product.new
+
+    render :new
+  end
+
+  def create
+    Product.create(create_product_params[:product])
+
+    redirect_to controller: "vendors", action: "products", id: params[:product][:vendor_id]
```

Comment:
Great job finding a way to direct to a path through the vendors controller. Is there a way you could do the same thing with a shorter line?